sam/cigar.go

```diff
@@ -197,6 +196,21 @@ func init() {
 	}
 }
 
+var powers = []int{1, 10, 100, 1000, 10000, 100000, 1000000, 10000000, 100000000}
+
+// atoi is an internal function; bytes in b are known to be digits.
+func atoi(b []byte) (int, error) {
+	n := 0
+	k := len(b) - 1
+	for i, v := range b {
+		n += int(v-'0') * powers[k-i]
+	}
+	if n >= 1<<28 || n < 0 {
```

Comment:
`n < 0 || 1<<28 <= n`. This says n outside 0 to 1<<28 very clearly, so is preferred.

There is no test for invalid CIGAR handling. I'll send a PR for that.